server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1760,8 +1761,22 @@ public void updateSegmentMetadata(final Set<DataSegment> segments)
           @Override
           public Void inTransaction(Handle handle, TransactionStatus transactionStatus) throws Exception
           {
+            final String updatePayload = StringUtils.format("UPDATE %s SET payload = :payload WHERE id = :id", dbTables.getSegmentsTable());
+            final PreparedBatch batch = handle.prepareBatch(updatePayload);
+
             for (final DataSegment segment : segments) {
-              updatePayload(handle, segment);
+              String id = segment.getId().toString();
+              byte[] payload = jsonMapper.writeValueAsBytes(segment);
+
+              batch.bind("id", id).bind("payload", payload).add();
+            }
+
+            try {
+              batch.execute();
+            }
+            catch (DBIException e) {
+              log.error(e, "Exception inserting into DB");
```

Comment:
```suggestion
              log.error(e, "Exception updating segment metadata in DB");
```

Reply:
Would be nice to also mention the datasource and number of segments we were trying to update in this log message.